tracing-appender/src/rolling/compression.rs

```diff
@@ -0,0 +1,217 @@
+//! Defines configuration for passing compression options
+//!
+//! Currently only gzip compression is implemented.
+use flate2::Compression;
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) enum GzipCompressionLevelLiteral {
+    None,
+    Fast,
+    Best,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+#[repr(u32)]
+pub(crate) enum GzipCompressionLevelNumerical {
+    Level0 = 0,
+    Level1 = 1,
+    Level2 = 2,
+    Level3 = 3,
+    Level4 = 4,
+    Level5 = 5,
+    Level6 = 6,
+    Level7 = 7,
+    Level8 = 8,
+    Level9 = 9,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) enum GzipCompressionLevel {
+    Literal(GzipCompressionLevelLiteral),
+    Numerical(GzipCompressionLevelNumerical),
+}
+
+/// Defines a conversion between `CompressionOption` and `flate2::Compression`
+impl Into<Compression> for GzipCompressionLevel {
+    fn into(self) -> Compression {
+        match self {
+            GzipCompressionLevel::Literal(lit) => match lit {
+                GzipCompressionLevelLiteral::None => Compression::none(),
+                GzipCompressionLevelLiteral::Fast => Compression::fast(),
+                GzipCompressionLevelLiteral::Best => Compression::best(),
+            },
+            GzipCompressionLevel::Numerical(num) => Compression::new(num as u32),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) struct GzipCompression {
+    pub(crate) level: GzipCompressionLevel,
+}
+
+/// Data structure to pass compression parameters
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) enum CompressionConfig {
+    Gzip(GzipCompression),
+}
+
+impl CompressionConfig {
+    pub(crate) fn gz_compress_level(&self) -> Compression {
+        match self {
+            CompressionConfig::Gzip(gz) => {
+                let level = gz.level.clone().into();
+                level
```

Comment:
We don't need the variable here either cause we are just returning it

Reply:
Fixed will push in a moment.